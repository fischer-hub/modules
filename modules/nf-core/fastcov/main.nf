process FASTCOV {
    tag "$meta.id"
    label 'process_low'

    // This tool is not available on conda
    //conda "${moduleDir}/environment.yml"
    container "docker://raverjay/fastcov:0.1.3--ba8c8cf6ae19"

    input:
    tuple val(meta), path(bam), path(index)

    output:
    tuple val(meta), path("${meta.id}.${task.ext.file_ext ?: 'png'}"), emit: coverage_plot
    path "versions.yml"                                              , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def file_ext = task.ext.file_ext ?: 'png'
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '0.1.3' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    fastcov.py -o ${prefix}.${file_ext} ${args} ${bam}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fastcov: $VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def file_ext = task.ext.file_ext ?: 'png'
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '0.1.3' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.

    """
    touch ${prefix}.${file_ext}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fastcov: $VERSION
    END_VERSIONS
    """
}
