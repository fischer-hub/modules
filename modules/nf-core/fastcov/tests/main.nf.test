nextflow_process {

    name "Test Process FASTCOV"
    script "../main.nf"
    process "FASTCOV"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "fastcov"

    test("fastcov single input bam") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true) ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success }
            )
        }

    }

    test("fastcov - stub") {

        options "-stub"

        when {
            process {
                """                
                input[0] = [
                    [ id:'test'], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true) ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
