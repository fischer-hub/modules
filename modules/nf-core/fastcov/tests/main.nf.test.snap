{
    "fastcov single input bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.png:md5,d4fe558c167986a7e1423db6d48bbf04"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1b47585c2aa870d7ffad63209dfe6d2b"
                ],
                "coverage_plot": [
                    [
                        {
                            "id": "test"
                        },
                        "test.png:md5,d4fe558c167986a7e1423db6d48bbf04"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1b47585c2aa870d7ffad63209dfe6d2b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T12:46:42.907171276"
    },
    "fastcov - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1b47585c2aa870d7ffad63209dfe6d2b"
                ],
                "coverage_plot": [
                    [
                        {
                            "id": "test"
                        },
                        "test.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1b47585c2aa870d7ffad63209dfe6d2b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T12:04:31.687126039"
    }
}